src/sage/modules/free_module_pseudohomspace.py

```diff
@@ -0,0 +1,338 @@
+            self._derivation = ore._derivation
+        else:
+            self._morphism = self._derivation = None
+        ring = ore.base_ring()
+        self._matrix_space = MatrixSpace(ring, domain.dimension(), codomain.dimension())
+
+    def _element_constructor_(self, f, side="left"):
+        r"""
+        Return the element of this parent constructed from the
+        given data.
+
+        TESTS::
+
+            sage: F.<z> = GF(5^3)
+            sage: Frob = F.frobenius_endomorphism()
+            sage: V = F^2
+            sage: H = V.pseudoHom(Frob)
+
+            sage: H([[1, z], [z, z^2]])
+            Free module pseudomorphism (twisted by z |--> z^5) defined by the matrix
+            [  1   z]
+            [  z z^2]
+            Domain: Vector space of dimension 2 over Finite Field in z of size 5^3
+            Codomain: Vector space of dimension 2 over Finite Field in z of size 5^3
+        """
+        return self.element_class(self, f, side)
+
+    def __reduce__(self):
+        r"""
+        TESTS::
+
+            sage: F = GF(125)
+            sage: Frob = F.frobenius_endomorphism()
+            sage: M = F^2
+            sage: H = M.pseudoHom(Frob)
+            sage: loads(dumps(M)) is M
+            True
+        """
+        if self._derivation is None:
+            twist = self._morphism
+        else:
+            twist = self._derivation
+        return FreeModulePseudoHomspace, (self.domain(), self.codomain(), twist)
+
+    def _repr_(self):
+        r"""
+        Returns a string representation of this pseudomorphism space.
+
+        EXAMPLES::
+
+            sage: Fq = GF(7^3)
+            sage: Frob = Fq.frobenius_endomorphism()
+            sage: V = Fq^2
+            sage: V.pseudoHom(Frob)  # indirect doctest
+            Set of Pseudoendomorphisms (twisted by z3 |--> z3^7) of
+            Vector space of dimension 2 over Finite Field in z3 of size 7^3
+
+        ::
+
+            sage: V.pseudoHom(Frob, codomain=Fq^3)  # indirect doctest
+            Set of Pseudomorphism (twisted by z3 |--> z3^7)
+            from Vector space of dimension 2 over Finite Field in z3 of size 7^3
+            to Vector space of dimension 3 over Finite Field in z3 of size 7^3
+
+        ::
+
+            sage: A.<t> = QQ[]
+            sage: d = A.derivation()
+            sage: M = A^3
+            sage: M.pseudoHom(d)
+            Set of Pseudoendomorphisms (twisted by d/dt) of Ambient free module of rank 3 over
+            the principal ideal domain Univariate Polynomial Ring in t over Rational Field
+        """
+        twist = self._ore._repr_twist()
+        if self.domain() is self.codomain():
+            return "Set of Pseudoendomorphisms (%s) of %s" % (twist, self.domain())
+        else:
+            return "Set of Pseudomorphism (%s) from %s to %s" % (twist, self.domain(), self.codomain())
+
+    def ore_ring(self, var='x'):
+        r"""
+        Return the underlying Ore polynomial ring.
+
```

Comment:
Suggestion of adding that:

```suggestion
        There exists an Ore polynomial ring `A` such that any pseudo
        morphism in ``self`` can be seen as an `A`-linear application.
        This method returns it.

```
```

Reply:
I change a bit the phrasing because actually pseudomorphisms are not $A$-linear.